src/monmove.cpp

```diff
@@ -1723,13 +1736,9 @@ void monster::knock_back_to( const tripoint &to )
 
     // If we're still in the function at this point, we're actually moving a tile!
     if( g->m.has_flag_ter( TFLAG_DEEP_WATER, to ) ) {
-        if( g->m.has_flag( "LIQUID", to ) && can_drown() ) {
-            die( nullptr );
-            if( u_see ) {
-                add_msg( _( "The %s drowns!" ), name() );
-            }
-
-        } else if( has_flag( MF_AQUATIC ) ) { // We swim but we're NOT in water
+        // die_if_drowning will kill the monster if necessary, but if the deep water
+        // tile is on a vehicle, we should check for swimmers out of water
+        if( !die_if_drowning( to, 1 ) && has_flag( MF_AQUATIC ) ) {
```

Comment:
Nitpick, but chance = 1 is default.
```suggestion
        if( !die_if_drowning( to ) && has_flag( MF_AQUATIC ) ) {
```